management/testutils/managementclient.go

```diff
@@ -27,5 +27,62 @@ func GetTestClient(t *testing.T) management.Client {
 	if err != nil {
 		t.Fatal(err)
 	}
-	return client
+	return logger{client, t}
+}
+
+type logger struct {
+	management.Client
+	*testing.T
```

Comment:
if you're not gonna let others do `client.Fatalf`, I think this should be privately embedded (i.e. `t *testing.T`).